File: exercise/java/day07/src/test/java/ci/CapturingLogger.java
package ci;

import ci.dependencies.Logger;

import java.util.ArrayList;
import java.util.List;

public class CapturingLogger implements Logger {
    private final List<String> lines = new ArrayList<>();

    @Override
    public void info(String message) {
        lines.add("INFO: " + message);
    }

    @Override
    public void error(String message) {
        lines.add("ERROR: " + message);
    }

    public List<String> getLoggedLines() {
        return lines;
    }
}

File: exercise/ts/day07/tests/capturingLogger.ts
import {Logger} from "../src/ci";

export class CapturingLogger implements Logger {
    private lines: string[] = [];

    info(message: string): void {
        this.lines.push(`INFO: ${message}`)
    }

    error(message: string): void {
        this.lines.push(`ERROR: ${message}`)
    }

    loggedLines = () => this.lines;
}